test/runtests.jl

```diff
@@ -316,16 +316,64 @@ end
 
         t = Tables.Schema((:a, :y, :z), (Int32, String, Any))
         for s in (GrandchildV1SchemaVersion(), ChildV1SchemaVersion(), ParentV1SchemaVersion())
-            @test_throws ArgumentError("could not find expected field `x` in $t") Legolas.validate(t, s)
+            msg = """
+                  Schema violation(s) found for Tables.Schema:
+                   :a  Int32
+                   :y  String
+                   :z  Any
+                   - Could not find expected field: `x`
+                  """
+            @test_throws ArgumentError(msg) Legolas.validate(t, s)
             @test !Legolas.complies_with(t, s)
             @test isequal(Legolas.find_violation(t, s), :x => missing)
+            @test isequal(Legolas.find_violations(t, s), [:x => missing])
         end
 
-        t = Tables.Schema((:x, :a, :y), (ComplexF64, Int32, String))
+        # Multiple missing field violations
+        t = Tables.Schema((:a, :z), (Int32, Any))
         for s in (GrandchildV1SchemaVersion(), ChildV1SchemaVersion(), ParentV1SchemaVersion())
-            @test_throws ArgumentError("field `x` has unexpected type; expected <:$(Vector), found $(Complex{Float64})") Legolas.validate(t, s)
+            msg = """
+                  Schema violation(s) found for Tables.Schema:
+                   :a  Int32
+                   :z  Any
+                   - Could not find expected field: `x`
+                   - Could not find expected field: `y`
+                  """
```

Comment:
Looking at this error message in isolation I think we could provide more context yet:
```suggestion
            msg = """
                  Tables.Schema violates Legolas schema $s:
                    - Could not find expected field: `x`
                    - Could not find expected field: `y`
                  Provided Tables.Schema:
                   :a  Int32
                   :z  Any
                  """
```

Reply:
updated, and updated top level PR comment to reflect new format.